lib/calculate-all.rb

```diff
@@ -1,89 +1,84 @@
-require "calculate-all/version"
-require "active_record"
+require 'calculate-all/version'
+require 'calculate-all/helpers'
+require 'calculate-all/querying'
 
 module CalculateAll
+  # Method to aggregate function results in one request
   def calculate_all(*function_aliases, **functions, &block)
+
+    # If only one function_alias is given, the result can be just a single value
+    # So return [{ cash: 3 }] instead of [{ cash: { count: 3 }}]
     if function_aliases.size == 1 && functions == {}
       return_plain_values = true
     end
+
+    # Convert the function_aliases to actual SQL
     functions.merge!(
       CalculateAll::Helpers.decode_function_aliases(function_aliases)
     )
+
+    # Check if any functions are given
     if functions == {}
-      raise ArgumentError, "provide at least one function to calculate"
-    end
-    # groupdate compatibility
-    group_values = self.group_values
-    if !group_values.is_a?(Array) && group_values.respond_to?(:relation)
-      group_values = group_values.relation
+      raise ArgumentError, 'provide at least one function to calculate'
     end
+
+    # If function is called without a group, the pluck method will still return
+    # an array but it is an array with the final results instead of each group
+    # The plain_rows boolean states how the results should be used
     if functions.size == 1 && group_values.size == 0
       plain_rows = true
     end
 
+    # Final output hash
     results = {}
 
-    pluck(*group_values, *functions.values).each do |row|
+    # Fetch all the requested calculations from the database
+    # Note the map(&:to_s). It is required since groupdate returns a
```

Comment:
I'm not really sure that won't break some complex groupings.
But, hey, I didn't test for them, and I'm not an active user of my own library anymore (unlike, it seems, you), so I'm going to accept that pull request half-blindly, and will wait for a bug report, if any.

It also looks like gem still works with rails 4 and groupdate 2.